### ISSUE TITLE
Reproduction of Stage 1

### ISSUE THREAD

**Issue by luckyyf0517** [ZechengLi19/Uni-Sign#8]
Hello,

Thank you for your outstanding work on this project.     I have been attempting to replicate the pretraining process (Stage 1) on the CSL-News dataset using pose data exclusively.     For my downstream task, 3D pose data is required, so I have re-annotated the image data using RTMPose3D.

Due to server storage limitations, I have only utilized approximately 50% of the data.     Additionally, according to Table 8 in your paper, using half of the dataset seemed sufficient to achieve good results.

However, when I conducted training using the default configuration of Stage 1 from the unisign code and compared the results with the `csl_stage1_weight.pth` on the CSL-News dataset, I found that I was unable to match the precision of the pretrained weights you provided.

I have a couple of questions:
1. During the training process for `csl_stage1_weight.pth`, were the hyperparameters identical to those provided in the code (such as learning rate, label smoothing, max length of segments)?

2. I noticed that the way train/test splits are done in the dataset is by taking the last 1% from the JSON file as the test set.     Yet, when I evaluated both the training and test sets on CSL-News, I found the performance metrics as follows:

**Training Set Metrics:**
```plaintext
BLEU Scores:
- BLEU1: 55.67
- BLEU2: 47.72
- BLEU3: 42.36
- BLEU4: 38.36
ROUGE-L F1: 56.35
```

**Test Set Metrics:**
```plaintext
BLEU Scores:
- BLEU1: 63.05
- BLEU2: 55.36
- BLEU3: 49.88
- BLEU4: 45.75
ROUGE-L F1: 63.08
```

I noticed the metrics on the test set are even higher than those on the training set.     Given the overfitting issues during model training, this result seems quite strange.

I understand that CSL-News is mainly used as a pretraining dataset and the evaluation metrics on downstream tasks (such as CSL-Daily/How2Sign) could be more indicative.     However, comparing my reproduction of the pretraining stage with `csl_stage1_weight.pth` on CSL-News is meaningful for me, even though the paper does not mention evaluation results on CSL-News.     My hypothesis is: potentially, the data split for pretraining `csl_stage1_weight.pth` might differ from the current code, or maybe the entire CSL-News dataset was used during pretraining?   If that’s the case, I may stop focusing on the Stage 1 metrics.

Thank you for your insights!

**Comment by ZechengLi19** [comment]
Thank you for your interest in Uni-Sign. 😀

Regarding your first question, the hyperparameters in this repository are identical to those used in the paper experiments. In fact, the hyperparameters for stages 1, 2, and 3 match exactly with what I used during the experiments. However, it's important to note that since the released code has been cleaned, there may be slight performance discrepancies. To address this, I reproduced the results on CSL-Daily and found that the performance does not deviate significantly from what was reported in the paper.

As for your second question, I suspect there might be an issue with your performance evaluation. Please double-check your code. If you choose to use `csl_stage1_weight.pth` for inference, ensure that the test data you're using is consistent with the last 1% of the original JSON file. Otherwise, you may observe inflated performance metrics. Based on my experiment, the BLEU4 of `csl_stage1_weight.pth` on the test set of CSL-News is approximately 21. (PS: The JSON file used in my original experiments is the same as the released one.) 

Additionally, it's worth mentioning that during the experiments, I discovered that the performance on CSL-News does not directly reflect downstream performance. In other words, improvements in the performance on CSL-News do not inevitably lead to enhancements in downstream tasks.

**Comment by luckyyf0517** [comment]
Thank you for your quick and helpful response.

After re-evaluating my code, I realized that I mistakenly did not iterate over the complete dataset, which led to abnormal metrics on smaller samples.

I have now moved forward to fine-tune the model with the daily data, and everything is progressing well. Using a model pretrained on 25% of the csl-news dataset, I achieved results close to those reported in the paper.

I believe my issues have been resolved. Thank you once again for your assistance.
